demoapps/passwordlogin/java/org/openyolo/demoapp/passwordlogin/ObservableViewModel.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2017 The OpenYOLO Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openyolo.demoapp.passwordlogin;
+
+import android.app.Application;
+import android.arch.lifecycle.AndroidViewModel;
+import android.databinding.Observable;
+import android.databinding.PropertyChangeRegistry;
+import android.support.annotation.NonNull;
+
+/**
+ * A base class for use by view models that utilize Android data binding.
+ */
+public abstract class ObservableViewModel extends AndroidViewModel implements Observable {
```

Comment:
In order to be able to bind a value in a layout, it must be an Observable. AndroidViewModel is not an observable, so I had to create a subclass which implements this to allow data binding. As this composition of view model and observable is reusable (it never really changes from view to view), it was worth creating a base type for extension by my other views, rather than extending AndroidViewModel and copying this implementation into every type.